web/src/components/repo/settings/SecretsTab.vue

```diff
@@ -81,14 +82,22 @@ export default defineComponent({
     const selectedSecret = ref<Partial<Secret>>();
     const isEditingSecret = computed(() => !!selectedSecret.value?.id);
 
-    async function loadSecrets() {
+    async function loadSecrets(page: number): Promise<boolean> {
```

Comment:
Why async function need promise in return?

Reply:
Oh, ok, not used to typescript so much, looks strange tho but you are right, this is correct